framework/console/controllers/BaseMigrateController.php

```diff
@@ -179,14 +179,22 @@ public function actionUp($limit = 0)
 
         $applied = 0;
         if ($this->confirm('Apply the above ' . ($n === 1 ? 'migration' : 'migrations') . '?')) {
-            foreach ($migrations as $migration) {
-                if (!$this->migrateUp($migration)) {
-                    $this->stdout("\n$applied from $n " . ($applied === 1 ? 'migration was' : 'migrations were') ." applied.\n", Console::FG_RED);
-                    $this->stdout("\nMigration failed. The rest of the migrations are canceled.\n", Console::FG_RED);
+            try {
+                foreach ($migrations as $migration) {
+                    if (!$this->migrateUp($migration)) {
+                        $this->stdout("\n$applied from $n " . ($applied === 1 ? 'migration was' : 'migrations were') ." applied.\n", Console::FG_RED);
```

Comment:
Shouldn't it be replaced with `showSummaryOnError()` call?

Reply:
Do you mean to extract full `try` to method with which arguments?